Kianoosh76/IBS
# Refactor/ibs 75 migrate mark staff

## Description
* Jira: https://ibs-utm.atlassian.net/browse/IBS-75
* Refactored Mark endpoints for Staff.

## Changes Made
* Refactored following endpoints:
    * ```module/mark/staff/all_csv.js```
    * ```module/mark/staff/all.js```
    * ```module/mark/staff/hide.js```
    * ```module/mark/staff/is_hidden.js```
    * ```module/mark/staff/one.js```
    * ```module/mark/staff/release.js```
    * ```module/mark/staff/submit.js```
    * ```module/mark/staff/upload.js```
* Added csv test files in:
    * ```module/tests/test-data/```
   
## Test Guidelines
In /backend/ run the following and ensure that they all passes:
```bash
npm test mark_staff_all
```
```bash
npm test mark_staff_csv_all
```
```bash
npm test mark_staff_hide
```
```bash
npm test mark_staff_is_hidden
```
```bash
npm test mark_staff_one
```
```bash
npm test mark_staff_release
```
```bash
npm test mark_staff_submit
```
```bash
npm test mark_staff_upload
```

## Checklist

### For PR author:
- [ ] Reviewed code changes.
- [ ] Added appropriate comments.
- [ ] Ensured consistent code style and formatting.
- [ ] Included unit/integration tests.
- [ ] Handled edge cases appropriately.
- [ ] Celebrated accomplishments 🎉
### For PR reviewer:
- [ ] Reviewed code changes.
- [ ] Verified clear and helpful comments.
- [ ] Ensured code style aligns with standards.
- [ ] Examined test coverage.
- [ ] Provided feedback on potential edge cases
